packages/suite-desktop/src-electron/preload.ts

```diff
@@ -1,24 +1,43 @@
 import { contextBridge, ipcRenderer } from 'electron';
 
-const validChannels = ['restart-app', 'start-bridge', 'oauth-receiver', 'oauth'];
+// TODO: Move to exported constant (to be shared)
+const validChannels = [
+    'restart-app',
+    'start-bridge',
+    'oauth-receiver',
+    'oauth',
+    // Events
+    'update/checking',
+    'update/available',
+    'update/not-available',
+    'update/error',
+    'update/downloading',
+    'update/downloaded',
+    'update/skip',
+];
 
 contextBridge.exposeInMainWorld('desktopApi', {
     send: (channel: string, data?: any) => {
-        // whitelist channels
         if (validChannels.includes(channel)) {
             ipcRenderer.send(channel, data);
         }
     },
     on: (channel: string, func: Function) => {
         if (validChannels.includes(channel)) {
-            // @ts-ignore: event value not used on purpose
-            ipcRenderer.on(channel, (event, ...args) => func(...args));
+            ipcRenderer.on(channel, (_, ...args) => func(...args));
         }
     },
     off: (channel: string, func: Function) => {
         if (validChannels.includes(channel)) {
-            // @ts-ignore: event value not used on purpose
-            ipcRenderer.off(channel, (event, ...args) => func(...args));
+            ipcRenderer.off(channel, (_, ...args) => func(...args));
         }
     },
+    // App ready
+    ready: () => ipcRenderer.send('ready'),
```

Comment:
validChannels checks are missing in all new methods

Reply:
This is done on purpose. I've decided to use dedicated methods for send calls and have only `on` and `off` use the channel array. Moving forward, I think it's a better pattern to follow.

Reply:
ahh i see it now! great